catalog/transformer/catalog-transformer-generic-impl/src/main/java/ddf/catalog/transformer/generic/xml/impl/SaxEventHandlerDelegate.java

```diff
@@ -0,0 +1,136 @@
+ */
+package ddf.catalog.transformer.generic.xml.impl;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.transformer.generic.xml.SaxEventHandler;
+
+public class SaxEventHandlerDelegate extends DefaultHandler {
+
+    private static SAXParser parser;
+
+    private List<SaxEventHandler> eventHandlers;
+
+    private String namespace;
+
+    private InputStream stream;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SaxEventHandlerDelegate.class);
+
+    public SaxEventHandlerDelegate() {
+        try {
+            // Read set up
+            SAXParserFactory factory = SAXParserFactory.newInstance();
+
+            factory.setSchema(null);
+            factory.setNamespaceAware(true);
+            factory.setValidating(false);
+            //            factory.setFeature("http://xml.org/sax/features/namespaces", false);
+            factory.setFeature("http://xml.org/sax/features/validation", false);
+            factory.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar",
+                    false);
+            factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd",
+                    false);
+
+            parser = factory.newSAXParser();
+        } catch (Exception e) {
+            LOGGER.debug("Exception thrown during creation of SaxEventHandlerDelegate. Probably caused by one of the setFeature calls", e);
+        }
+    }
+
+    public SaxEventHandlerDelegate(String namespace, SaxEventHandler... eventHandlers) {
+        this();
+        this.namespace = namespace;
+        this.eventHandlers = Arrays.asList(eventHandlers);
+    }
+
+    public SaxEventHandlerDelegate(SaxEventHandler... eventHandlers) {
+        this();
+        this.eventHandlers = Arrays.asList(eventHandlers);
+    }
+
+    public SaxEventHandlerDelegate(List<SaxEventHandler> eventHandlers) {
+        this();
+        this.eventHandlers = eventHandlers;
+    }
+
+    public Metacard read(InputStream inputStream) {
+        Metacard metacard = new MetacardImpl();
+        try {
+            stream = inputStream;
+            parser.parse(new BufferedInputStream(inputStream), this);
+        } catch (IOException | SAXException e) {
+            LOGGER.debug("Exception thrown during parsing of inputStream", e);
+        }
+
+        // Populate metacard with all attributes constructed in SaxEventHandlers during parsing
+        for (SaxEventHandler eventHandler : eventHandlers) {
```

Comment:
There should be a new delegate for each request and each request will be new event handlers too.